code/modules/mob/living/life.dm

```diff
@@ -152,4 +152,29 @@
 	var/grav_strength = gravity - GRAVITY_DAMAGE_THRESHOLD
 	adjustBruteLoss(min(GRAVITY_DAMAGE_SCALING * grav_strength, GRAVITY_DAMAGE_MAXIMUM) * seconds_per_tick)
 
+/// Returns if this mob is breathing at this moment. Fails if they cannot breathe period, or if they are dead.
+/mob/living/proc/currently_breathing()
+	if (!can_breathe())
+		return FALSE
+	if (stat == DEAD)
+		return FALSE
+	return TRUE
+
+/// Returns if this mob can breathe at all. Fails if they have NOBREATH or do not have the breathing trait.
+/mob/living/proc/can_breathe()
+	if (!HAS_TRAIT(src, TRAIT_BREATHES))
```

Comment:
Why should this be a trait at all? Shouldn't we assume they can breathe unless they have nobreath?

Reply:
nope, because things w/o nobreath dont breathe (basicmobs, silicons, simplemobs)